### PR TITLE
ipu6: update packages

### PR DESCRIPTION
**Pull request by matejc**
## Things done

Part of https://github.com/NixOS/nixpkgs/issues/225743

To test out:

```
      hardware.ipu6 = {
        enable = true;
        platform = "ipu6ep";
      };
      boot.kernelPackages = pkgs.linuxPackages_6_11;
```

I tested this on kernel 6.11.1 and 6.11.2

It works on X11 for Chromium well enough, but not Firefox

Works on Wayland on Chromium, but not Firefox, although Chromium is puking continuously in terminal:

```
[35650:35723:1011/153603.111515:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= YUV_420_BIPLANAR and usage=SCANOUT_CPU_READ_WRITE
[35650:35723:1011/153603.111741:ERROR:gpu_channel.cc(502)] Buffer Handle is null.
[35731:16:1011/153603.112160:ERROR:shared_image_interface_proxy.cc(134)] Buffer handle is null. Not creating a mailbox from it.
[35650:35723:1011/153603.145283:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= YUV_420_BIPLANAR and usage=SCANOUT_CPU_READ_WRITE
[35650:35723:1011/153603.145592:ERROR:gpu_channel.cc(502)] Buffer Handle is null.
[35731:16:1011/153603.145922:ERROR:shared_image_interface_proxy.cc(134)] Buffer handle is null. Not creating a mailbox from it.
[35650:35723:1011/153603.179060:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= YUV_420_BIPLANAR and usage=SCANOUT_CPU_READ_WRITE
[35650:35723:1011/153603.179321:ERROR:gpu_channel.cc(502)] Buffer Handle is null.
[35731:16:1011/153603.179675:ERROR:shared_image_interface_proxy.cc(134)] Buffer handle is null. Not creating a mailbox from it.
[35650:35723:1011/153603.213402:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= YUV_420_BIPLANAR and usage=SCANOUT_CPU_READ_WRITE
[35650:35723:1011/153603.213568:ERROR:gpu_channel.cc(502)] Buffer Handle is null.
[35731:16:1011/153603.213713:ERROR:shared_image_interface_proxy.cc(134)] Buffer handle is null. Not creating a mailbox from it.
```

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by matejc** [comment]
Also works in obs-studio and Slack for desktop (using Wayland).

**Review by flokli — APPROVED** [review]
I can confirm this brings back video in Chromium. I'm ok with merging this once the comment is fixed. Nice job!

For completeness, firefox logs the following:

```
Okt 11 17:19:03 x1 pipewire[3065]: spa.v4l2: '/dev/video0' VIDIOC_REQBUFS Invalid argument
Okt 11 17:19:03 x1 pipewire[3065]: pw.port: 0x5628da3bfe40: negotiate buffers on node: -22 (Invalid argument)
Okt 11 17:19:03 x1 pipewire[3065]: pw.port: 0x5628da3bfe40: state ready -> error (can't negotiate buffers on port)
Okt 11 17:19:03 x1 pipewire[3065]: pw.link: (196.0.0 -> 211.0.0) allocating -> error (can't negotiate buffers on port) (error-ready)
```

File: nixos/modules/hardware/video/webcam/ipu6.nix [hunk 1]
```diff
@@ -27,8 +27,7 @@ in
   config = mkIf cfg.enable {
 
     # Module is upstream as of 6.10
-    boot.extraModulePackages = with config.boot.kernelPackages;
-      optional (kernelOlder "6.10") ipu6-drivers;
+    boot.extraModulePackages = with config.boot.kernelPackages; [ ipu6-drivers ];
```
[the remark this answers is not in the document]
Comment (matejc) [reply nixos/modules/hardware/video/webcam/ipu6.nix:30]: At least ipu6's psys module is not in-tree, its missing from kernel

[the remark this answers is not in the document]
Comment (flokli) [reply nixos/modules/hardware/video/webcam/ipu6.nix:30]: This brings in the following modules:

```
./lib/modules/6.11.3/updates/drivers/media/i2c/hi556.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/hm11b1.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/hm2170.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/hm2172.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/ov01a10.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/ov01a1s.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/ov02c10.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/ov02e10.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/ov05c10.ko.xz
./lib/modules/6.11.3/updates/drivers/media/i2c/ov2740.ko.xz
./lib/modules/6.11.3/updates/drivers/media/pci/intel/ipu6/psys/intel-ipu6-psys.ko.xz
```

So I guess we can update the comment to say: 

```
# Still needs various out-of-tree i2c and the `intel-ipu6-psys` kernel driver
```

?

[the remark this answers is not in the document]
Comment (matejc) [reply nixos/modules/hardware/video/webcam/ipu6.nix:31]: Ok, thats better than what I committed just now

[the remark this answers is not in the document]
Comment (matejc) [reply nixos/modules/hardware/video/webcam/ipu6.nix:31]: Pushed, check again please

**Comment by flokli** [comment]
Now `linuxPackages.ipu6-drivers` fails to build.

We should probably update the `broken` attribute so it won't try (and fail to build) on 6.6.

**Comment by flokli** [comment]
I updated `pkgs/os-specific/linux/ipu6-drivers/default.nix` (the `meta.broken` attribute specifically).

**Comment by matejc** [comment]
@flokli anything else needs to be done?

**Comment by flokli** [comment]
No, I just wanted to wait for the eval checks to pass 😆 

Thanks for the reminder!

**Pull request merged**